content/en/real_user_monitoring/guide/best-practices-tracing-native-ios-android-apps.md

```diff
@@ -0,0 +1,167 @@
+{{< tabs >}}
+{{% tab "iOS" %}}
+Use the [manual context propagation][1] to inject trace headers into network requests. Use `setActive()` to link child spans to a parent span ([API reference][2]).
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/ios?tab=swiftpackagemanagerspm#:~:text=(Optional)%20To%20distribute%20traces%20between%20your%20environments%2C%20for%20example%20frontend%20%2D%20backend%2C%20you%20can%20either%20do%20it%20manually%20or%20leverage%20our%20auto%20instrumentation.%20In%20both%20cases%2C%20network%20traces%20are%20sampled%20with%20an%20adjustable%20sampling%20rate.%20A%20sampling%20of%2020%25%20is%20applied%20by%20default
+[2]: https://swiftpackageindex.com/datadog/dd-sdk-ios/develop/documentation/datadogtrace/otspan/setactive/
+
+{{% /tab %}}
+{{% tab "Android" %}}
+
+Use [OkHttp parent span helpers][1] or [OpenTelemetry addParentSpan][2] to link spans across threads.
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/android/?tab=kotlin#okhttp
+[2]: /tracing/trace_collection/custom_instrumentation/android/otel/?tab=kotlin#:~:text=(Optional)%20Add%20local%20parent%20span%20to%20the%20span%20generated%20around%20the%20OkHttp%20request%20in%20RUM%3A
+
+{{% /tab %}}
+{{< /tabs >}}
+
+**Note**: Wrapping RUM resource spans under a native span works automatically if the parent span is on the same thread as the network call. Otherwise, use the provided helpers to set the parent span manually.
+
+### Profiling native application performance
+
+Instrumenting multiple native spans and linking them (using `setActive` on iOS or `activateSpan` on Android) allows you to profile key parts of your app. This helps you:
+- Understand how different methods and components interact
+- Break down performance bottlenecks
+- Gain actionable insights into app behavior
+
+Here's how to manually instrument a span in your mobile app using the Trace SDK:
+
+{{< tabs >}}
+{{% tab "iOS" %}}
+```swift
+let span = tracer.startSpan(operationName: "<span_name>")
+// ... code to measure ...
+span.finish()
+```
+See [more iOS examples][1].
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/ios?tab=swiftpackagemanagerspm
+
+{{% /tab %}}
+{{% tab "Android" %}}
+
+```kotlin
+val span = tracer.buildSpan("<span_name>").start()
+// ... code to measure ...
+span.finish()
+```
+See [more Android examples][1].
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/android/?tab=kotlin
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Sampling
+
+Sampling in native mobile tracing controls which spans and traces are ingested in Datadog, helping you balance visibility with data volume. When you manually instrument spans in your app, the ones that are sampled in go through the APM retention filters which determine which ones end up being displayed for analysis in the Datadog interface.
+
+### Sampling parameters
+
+The following sampling rate parameters control different aspects of data collection:
```

Comment:
I've seen sampling percentages expressed in different ways - for example, sometimes a system will expect a rate of 20% to be expressed as `20`, but other systems will expect `0.2`. Maybe this is already obvious to someone already using the Trace SDK, but if you think there might be some ambiguity, it might help to spell out explicitly how to express these percentages.